op-service/logpipe/go.go

```diff
@@ -0,0 +1,66 @@
+package logpipe
+
+import (
+	"bytes"
+	"encoding/json"
+	"log/slog"
+
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type rawGoJSONLog map[string]any
+
+type StructuredGoLogEntry struct {
+	Message string
+	Level   slog.Level
+	Fields  map[string]any
+}
+
+func ParseGoStructuredLogs(line []byte) LogEntry {
+	dec := json.NewDecoder(bytes.NewReader(line))
+	dec.UseNumber() // to preserve number formatting
+	var e rawGoJSONLog
```

Comment:
Decoding into a struct, with just the fields that you need, may be better, to avoid decoding map elements fully that are unexpected. Like the rust version does.